### PR TITLE
Fix #1623 for npm dependencies 

### LINKED ISSUE #1623: gulp missing gem specification with from-source install
Pivoting from #1617 installation failure, I attempted to install from source on a docker image with only postres10.  Going through the steps on the main README.md, I hit an error in the npm/gem installation task.  Unfortunately, I don't know Ruby so I can't offer any deeper insights than just this CLI error.

```
root@499a761fe182:/evalai# gulp dev:runserver
[13:23:46] Using gulpfile /var/lib/postgres/data/evalai/gulpfile.js
[13:23:46] Starting 'dev:runserver'...
[13:23:46] Starting 'dev'...
[13:23:46] Starting 'clean'...
[13:23:46] Finished 'clean' after 54 ms
[13:23:46] Starting 'css'...
[13:23:46] Starting 'js'...
[13:23:46] Finished 'js' after 11 ms
[13:23:46] Starting 'html'...
[13:23:46] Finished 'html' after 1.78 ms
[13:23:46] Starting 'images'...
[13:23:46] Starting 'vendorjs'...
[13:23:46] Finished 'vendorjs' after 228 ms
[13:23:46] Starting 'vendorcss'...
[13:23:46] Finished 'vendorcss' after 44 ms
[13:23:46] Starting 'fonts'...
[13:23:46] Starting 'configDev'...
[13:23:46] Finished 'configDev' after 3.58 ms

events.js:165
      throw er; // Unhandled 'error' event
      ^
Error: Gem sass is not installed.
```

To remedy this, I have attempted to install `sass` and `node-sass` with commands like `npm install node-sass`.

Unfortunately, this leaves no option for trying the EvalAI platform at this time: can't add challenges to deployed version #1590, can't run from docker #1617, can't install from source (this issue).

### PR DESCRIPTION
**Pull request by ezavesky**
- version update for gulp-karma/karma
- and optional text addition for missing global sass
- add gulp-cli for those who aren't using the tool regularly (in new environment)

Please give a second review as I'm not as familiar with gulp/node edits.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +142,18 @@ You can also use Docker Compose to run all the components of EvalAI together. Th
     ```
     gulp dev:runserver
     ```
+   
+    * *Optional* If you get an error about missing the sass gem, you may need to also install it globally.
```
Comment (gautamjajoo) [inline README.md]: @deshraj Would it be correct to add the above error here or in the FAQ section for developers (Issue #1130 )

Comment (spyshiv) [reply README.md]: @gautamjajoo you're right. We are not gonna put error instructions with setup instructions. We should mention the recommended node version, npm version, bower version, ruby-sass etc In very clear way. And if this is known bug then we should move it to FAQ as you mentioned.

**Comment by ezavesky** [comment]
Oops.  It looks like I should close/discard this PR as there is a more complete contribution on #1639 as a fix to both #1637 and #1628.  Please advise.

**Comment by deshraj** [comment]
> I have no problem creating a FAQ but #1130 is over a year old now. Should I just drop a new one in as docs/source/FAQ.md? 

Yeah that would be great. Thanks @ezavesky :) 

> Consolidate the setup directions in one place, link to one/them from the main page.
We already have all the consolidated setup instructions here: http://evalai.readthedocs.io/en/latest/setup.html#how-to-setup

We can go ahead and remove the detailed installation instructions from the README and add links to the installation instructions for specific platforms in the README. Does that sound good to you? 
 
> There is a precedent set for inline FAQ/error spotting in some setup directions already (https://github.com/Cloud-CV/EvalAI/blob/master/docs/source/setup.md#common-errors). Personally, I find those inline hints more useful than having to consult an additional FAQ (which wasn't linked anywhere and then requires users to (a) know about that document, and (b) search another document)

Yeah, I agree. We can have the installation related errors to be mentioned inline but at the same time I think adding FAQ would be great since we could point people to FAQ for other problems such as how to start contributing, how to setup EvalAI on AWS etc.

**Comment by gautamjajoo** [comment]
@deshraj @ezavesky Can I work on the issue #1130 becuase I have collected all the errors from gitter channel.

**Comment by deshraj** [comment]
@gautamjajoo yeah feel free to work on it. Please let us know if you want some more information. :)

**Pull request closed without merging**